rllib/agents/dreamer/dreamer.py

```diff
@@ -0,0 +1,254 @@
+import logging
+
+import random
+import numpy as np
+
+from ray.rllib.agents import with_common_config
+from ray.rllib.agents.dreamer.dreamer_torch_policy import DreamerTorchPolicy
+from ray.rllib.agents.trainer_template import build_trainer
+from ray.rllib.execution.common import STEPS_SAMPLED_COUNTER, \
+    LEARNER_INFO, _get_shared_metrics
+from ray.rllib.policy.sample_batch import SampleBatch
+from ray.rllib.policy.sample_batch import DEFAULT_POLICY_ID
+from ray.rllib.evaluation.metrics import collect_metrics
+from ray.rllib.agents.dreamer.dreamer_model import DreamerModel
+from ray.rllib.execution.rollout_ops import ParallelRollouts
+from ray.rllib.utils.typing import SampleBatchType
+
+logger = logging.getLogger(__name__)
+
+# yapf: disable
+# __sphinx_doc_begin__
+DEFAULT_CONFIG = with_common_config({
+    # PlaNET Model LR
+    "model_lr": 6e-4,
+    # Actor LR
+    "actor_lr": 8e-5,
+    # Critic LR
+    "critic_lr": 8e-5,
+    # Grad Clipping
+    "grad_clip": 100.0,
+    # Discount
+    "discount": 0.99,
+    # Lambda
+    "lambda": 0.95,
+    # Training iterations per data collection from real env
+    "dreamer_train_iters": 100,
+    # If specified, clip the global norm of gradients by this amount
```

Comment:
??

Reply:
Everytime one collects an episode from environment, one does 100 iterations of dreamer.

Reply:
No, :) I meant the comment: "If specified, clip the global norm of gradients by this amount", seems off here.